### PR TITLE
Check approval

### PR DESCRIPTION
**Pull request by stevenyyan**
### Motivation

It wasn't doing any recheck if the user has approved the token for spending once they approve the token for spending on one account and then change to another account. This results in not showing the approval button even when the user hasn't approved for the other account.

### Changes

- Recheck if the new account address has been approved upon changing accounts
- Other misc changes

### CODE REVIEW

File: components/General/index.tsx [hunk 1]
```diff
@@ -135,26 +135,31 @@ export const Section: React.FC<SProps> = styled(({ className, children, label, t
...
-    ticker: string;
+    ticker: string | undefined;
     clear?: boolean; // true then display outlined image
 }
 
 export const Logo: React.FC<LProps> = styled(({ className, ticker, clear }: LProps) => {
-    return <img className={className} src={clear ? clearLogos[ticker] : logos[ticker] ?? logos['TSLA']} alt="logo" />;
+    return (
+        <>
+            {(ticker === 'TSLA' || 'ETH' || 'LINK') && ticker !== undefined ? (
```
Comment (dospore) [inline components/General/index.tsx]: This boolean check is 
if (ticker === 'TSLA) or if ('ETH') or if ('LINK') && ticker !== undefined
my suggestion would be to create a ticker type in typescript and restrict ticker to ticker: TickerType | undefined
and then just have the ticker !== undefined

File: context/TracerContext.tsx [hunk 2]
```diff
@@ -69,6 +69,16 @@ export const SelectedTracerStore: React.FC<StoreProps> = ({ tracer, children }:
 
     const [tracerState, tracerDispatch] = useReducer(reducer, initialState);
 
+    // Reset approval state on account change
+    const resetApproval = async () => {
+        selectedTracer?.removeApproved(selectedTracer.address);
+        await selectedTracer?.checkApproved(account);
```
Comment (dospore) [inline context/TracerContext.tsx:74]: checkApproved already sets the approved values. you should be able to just call selectedTracer?.checkApproved(account) and remove the other two function calls

Comment (stevenyyan) [reply context/TracerContext.tsx:74]: That works. Thanks!

File: components/General/index.tsx (hunk 1 above)
Comment (stevenyyan) [reply components/General/index.tsx]: done

File: components/General/index.tsx [hunk 3]
```diff
@@ -135,26 +135,33 @@ export const Section: React.FC<SProps> = styled(({ className, children, label, t
...
     TSLA: '/img/logos/currencies/tesla.svg',
     ETH: '/img/logos/currencies/eth.svg',
-    TEST1: '/img/logos/currencies/eth.svg',
     LINK: '/img/logos/currencies/link.svg',
-    DEFAULT: '/img/logos/currencies/tesla.svg',
 };
 
+const tickerTypes = ['TSLA', 'ETH', 'LINK'];
+type TickerType = typeof tickerTypes[number];
+const isOfTickerType = (ticker: any): ticker is TickerType => tickerTypes.includes(ticker);
```
Comment (dospore) [inline components/General/index.tsx:151]:
```suggestion
type TickerType = 'TSLA' | 'ETH' | 'LINK';
```

File: components/General/index.tsx [hunk 4]
```diff
@@ -135,26 +135,33 @@ export const Section: React.FC<SProps> = styled(({ className, children, label, t
...
 interface LProps {
     className?: string;
-    ticker: string;
+    ticker: string | undefined;
```
Comment (dospore) [inline components/General/index.tsx:154]:
```suggestion
    ticker: TickerType | undefined;
```

File: components/General/index.tsx [hunk 5]
```diff
@@ -135,26 +135,33 @@ export const Section: React.FC<SProps> = styled(({ className, children, label, t
...
     clear?: boolean; // true then display outlined image
 }
-
 export const Logo: React.FC<LProps> = styled(({ className, ticker, clear }: LProps) => {
-    return <img className={className} src={clear ? clearLogos[ticker] : logos[ticker] ?? logos['TSLA']} alt="logo" />;
+    return (
+        <>
+            {isOfTickerType(ticker) && ticker !== undefined ? (
```
Comment (dospore) [inline components/General/index.tsx:160]:
```suggestion
            { ticker !== undefined ? (
```

File: components/General/index.tsx (hunk 3 above)
Comment (dospore) [inline components/General/index.tsx]:
```suggestion
const isOfTickerType = (ticker: string) => tickerTypes.includes(ticker);
```

File: components/General/index.tsx [hunk 6]
```diff
@@ -135,26 +135,33 @@ export const Section: React.FC<SProps> = styled(({ className, children, label, t
...
 
 const logos: Record<string, string> = {
     TSLA: '/img/logos/currencies/tesla.svg',
     ETH: '/img/logos/currencies/eth.svg',
-    TEST1: '/img/logos/currencies/eth.svg',
     LINK: '/img/logos/currencies/link.svg',
-    DEFAULT: '/img/logos/currencies/tesla.svg',
 };
 
+const tickerTypes = ['TSLA', 'ETH', 'LINK'];
```
Comment (dospore) [inline components/General/index.tsx]:
```suggestion
const tickerTypes: string[] = ['TSLA', 'ETH', 'LINK'];
```

**Review by dospore — CHANGES_REQUESTED** [review]
Leverage TypeScript here. It wont build if you dont pass it the correct type

**Review by dospore — APPROVED** [review]

**Pull request merged**